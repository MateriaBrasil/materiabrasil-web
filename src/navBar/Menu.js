import React, { Component, Fragment } from 'react'

import SignOut from '../auth/signOut/SignOut'
import Link, { SignIn } from './LinkMobile'
import './Link.css'

import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'

export default props => {
  const { anchorEl, open, handleClose } = props
  const { currentUser } = props
  const { firstName, lastName, suppliers } = currentUser || {}
  const name = `${firstName} ${lastName}`

  return (
    <Menu
      id="menu-appbar"
      className="menu-appbar"
      anchorEl={anchorEl}
      open={open}
      onClose={handleClose}
    >
      {currentUser ? (
        <Fragment>
          <MenuItem>
            {suppliers && suppliers.length > 0 ? (
              <Link
                to={`/suppliers/${suppliers[0].id}`}
                text={suppliers[0].name}
              />
            ) : (
              <Link to={`/suppliers/new`} text="Cadastrar fornecedor" />
            )}
          </MenuItem>
          <MenuItem>
            <Link to={`/profile`} text={name} />
          </MenuItem>
          <MenuItem>
            <SignOut {...props} />
          </MenuItem>
        </Fragment>
      ) : (
        <Fragment>
          <MenuItem onClick={handleClose}>
            <Link
              to={{
                pathname: '/auth/sign-up',
                state: { referrer: '/suppliers/new' },
              }}
              text="Cadastrar fornecedor"
            />
          </MenuItem>
          <MenuItem onClick={handleClose}>
            <SignIn {...props} />
          </MenuItem>
        </Fragment>
      )}
    </Menu>
  )
}

// {currentUser ? (
//   <Fragment>
//     {suppliers && suppliers.length > 0 ? (
//       <MenuItem>
//         <Link
//           to={`/suppliers/${suppliers[0].id}`}
//           text={suppliers[0].name}
//         />
//       </MenuItem>
//     ) : (
//       <MenuItem>
//         <Link to={`/suppliers/new`} text="Cadastrar fornecedor" />
//       </MenuItem>
//     )}
//     <MenuItem>
//       <Link to={`/profile`} text={name} />
//     </MenuItem>
//     <MenuItem>
//       <SignOut {...props} />
//     </MenuItem>
//   </Fragment>
// ) : (
//   <Fragment>
//     <MenuItem>
//       <Link
//         to={{
//           pathname: '/auth/sign-up',
//           state: { referrer: '/suppliers/new' },
//         }}
//         text="Cadastrar fornecedor"
//       />
//     </MenuItem>
//     <MenuItem>
//       <SignIn {...props} />
//     </MenuItem>
//   </Fragment>
// )}

// anchorOrigin={{
//   vertical: 'top',
//   horizontal: 'right',
// }}
// transformOrigin={{
//   vertical: 'top',
//   horizontal: 'right',
// }}
