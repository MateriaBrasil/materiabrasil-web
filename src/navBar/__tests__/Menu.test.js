import React from 'react'
import renderer from 'react-test-renderer'

import Menu from '../Menu'

jest.mock('react-router-dom/Link', () => props => (
  <div {...props}>Link - {props.children}</div>
))
jest.mock('@material-ui/core/Menu', () => props => (
  <div {...props}>Menu - {props.children}</div>
))
jest.mock('@material-ui/core/MenuItem', () => props => (
  <div {...props}>MenuItem - {props.children}</div>
))

jest.mock('../SignIn', () => props => <div {...props}>SignIn</div>)
jest.mock('../../auth/signOut/SignOut', () => props => (
  <div {...props}>SignOut</div>
))

it('renders correctly', () => {
  const tree = renderer.create(<Menu />)
  expect(tree).toMatchSnapshot()
})

describe('with current user', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Menu currentUser={{}} />)
    expect(tree).toMatchSnapshot()
  })

  describe('with suppliers', () => {
    it('renders correctly', () => {
      const suppliers = [{}, {}]

      const tree = renderer.create(<Menu currentUser={{ suppliers }} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
