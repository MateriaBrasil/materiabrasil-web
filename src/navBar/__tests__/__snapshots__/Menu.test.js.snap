// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  anchorEl={undefined}
  className="menu-appbar"
  id="menu-appbar"
  onClose={undefined}
  open={undefined}
>
  Menu - 
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/about"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Sobre
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/forum"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Fórum
      </h3>
    </div>
  </div>
  <div
    onClick={undefined}
  >
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/auth/sign-up",
          "state": Object {
            "referrer": "/suppliers/new",
          },
        }
      }
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Cadastrar fornecedor
      </h3>
    </div>
  </div>
  <div
    onClick={undefined}
  >
    MenuItem - 
    <div>
      SignIn
    </div>
  </div>
</div>
`;

exports[`with current user renders correctly 1`] = `
<div
  anchorEl={undefined}
  className="menu-appbar"
  id="menu-appbar"
  onClose={undefined}
  open={undefined}
>
  Menu - 
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/about"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Sobre
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/forum"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Fórum
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/suppliers/new"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Cadastrar fornecedor
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/profile"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        undefined undefined
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      currentUser={Object {}}
    >
      SignOut
    </div>
  </div>
</div>
`;

exports[`with current user with suppliers renders correctly 1`] = `
<div
  anchorEl={undefined}
  className="menu-appbar"
  id="menu-appbar"
  onClose={undefined}
  open={undefined}
>
  Menu - 
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/about"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Sobre
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/forum"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        Fórum
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/suppliers/undefined"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      />
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      className="navbar-link"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/profile"
    >
      Link - 
      <h3
        className="MuiTypography-root-1 MuiTypography-subheading-8 MuiTypography-colorInherit-20"
        style={
          Object {
            "alignSelf": "center",
            "display": "inline-block",
          }
        }
      >
        undefined undefined
      </h3>
    </div>
  </div>
  <div>
    MenuItem - 
    <div
      currentUser={
        Object {
          "suppliers": Array [
            Object {},
            Object {},
          ],
        }
      }
    >
      SignOut
    </div>
  </div>
</div>
`;
